api/service.go

```diff
@@ -19,7 +19,7 @@ import (
 )
 
 // registerServices registers all endpoints to the grpc server
-func registerServices(ctx context.Context, grpcServer *grpc.Server, gwmux *runtime.ServeMux, addr string, dopts []grpc.DialOption) error {
+func registerServices(ctx context.Context, config Config, grpcServer *grpc.Server, gwmux *runtime.ServeMux, addr string, dopts []grpc.DialOption) error {
```

Comment:
Question to @vimukthi-git @vedhavyas . According to the go context.Context it is meant to pass arbitrary information that is needed for the downstream processing. As we know we are passing the context pretty much from the beginning of the request, is it good practice to add the config there? instead of a separate attribute? What about other shared resources?

Reply:
I'm fine with either way, but seems `context.Context` is generally used for more dynamic info specific to a request or execution? I'm not quite sure. I'm fine with what you and @vedhavyas agrees on this.

Reply:
Thanks @vimukthi-git . I will instead wrap the `context.Context` in the centrifuge `Context` plus the config, so we can have a way to pass request specific data downstream.

Reply:
Exactly what I faced, to many castings downstream. Instead I am passing the typed centrifuge context.